logstash-core/lib/logstash/pipeline_action/create.rb

```diff
@@ -40,13 +40,19 @@ def execute(agent, pipelines)
           LogStash::Pipeline.new(@pipeline_config, @metric, agent)
         end
 
-      status = pipeline.start # block until the pipeline is correctly started or crashed
-
-      if status
-        pipelines[pipeline_id] = pipeline # The pipeline is successfully started we can add it to the hash
+      status = nil
+      pipelines.compute(pipeline_id) do |id,value|
+        status = pipeline.start # block until the pipeline is correctly started or crashed
```

Comment:
That's a good point. Theoretically, this shouldn't be possible because higher up, when we determine the actions to be run, we would do that check. Today that's safe because that code isn't concurrent. However, it may not be in the future.

I propose that we add an invariant that throws an exception if the pipeline does already exist, since that should never be the case.